packages/storage/src/indexed-db/index.ts

```diff
@@ -869,20 +869,26 @@ export class IndexedDb implements IndexedDbInterface {
     };
   }
 
-  async hasTokenBalance(addressIndex: AddressIndex, assetId: AssetId): Promise<boolean> {
-    const spendableNotes = await this.db.getAllFromIndex(
-      'SPENDABLE_NOTES',
-      'assetId',
-      uint8ArrayToBase64(assetId.inner),
-    );
+  async accountHasSpendableAsset(
+    { account }: PlainMessage<AddressIndex>,
+    { inner }: PlainMessage<AssetId>,
+  ): Promise<boolean> {
+    // asserts length of assetId.inner
+    bech32mAssetId({ inner });
 
-    return spendableNotes.some(note => {
-      const spendableNote = SpendableNoteRecord.fromJson(note);
-      return (
-        spendableNote.heightSpent === 0n &&
-        !isZero(getAmountFromRecord(spendableNote)) &&
-        spendableNote.addressIndex?.equals(addressIndex)
-      );
+    const assetNotes = (
+      await this.db.getAllFromIndex('SPENDABLE_NOTES', 'assetId', uint8ArrayToBase64(inner))
+    ).map(note => SpendableNoteRecord.fromJson(note));
+
+    const spendableNotes = assetNotes.filter(({ note, addressIndex, heightSpent }) => {
+      const noteAddressIndex = toPlainMessage(addressIndex ?? new AddressIndex());
+      const accountMatch = noteAddressIndex.account === account;
```

Comment:
is it correct to ignore account randomizer for this comparison?

Reply:
I don't think the randomizer is relevant here? https://github.com/penumbra-zone/web/blob/main/packages/wasm/crate/src/planner.rs#L177-L180.

cc @Valentine1898

Reply:
the randomizer should be ignored 
https://discord.com/channels/824484045370818580/1044457038886998137/1220144758970781707